sql-plugin/src/main/scala/org/apache/spark/sql/rapids/test/cpuJsonExpressions.scala

```diff
@@ -0,0 +1,317 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.test
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.{Objects, UUID}
+
+import scala.collection.mutable
+import scala.util.Random
+
+import ai.rapids.cudf.{ColumnVector, DType, HostColumnVector}
+import com.nvidia.spark.rapids.Arm.withResource
+import com.nvidia.spark.rapids.GpuColumnVector
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FSDataOutputStream, Path}
+
+import org.apache.spark.TaskContext
+import org.apache.spark.sql.catalyst.expressions.{GetJsonObject, Literal}
+import org.apache.spark.sql.types.StringType
+import org.apache.spark.unsafe.types.UTF8String
+
+case class CsvWriterWrapper(filePath: String, conf: Configuration) extends AutoCloseable {
+
+  // This is implemented as a method to make it easier to subclass
+  // ColumnarOutputWriter in the tests, and override this behavior.
+  private def getOutputStream: FSDataOutputStream = {
+    val hadoopPath = new Path(filePath)
+    val fs = hadoopPath.getFileSystem(conf)
+    fs.create(hadoopPath, false)
+  }
+  private var fileStream: FSDataOutputStream = getOutputStream
+
+  override def close(): Unit = {
+    if (fileStream != null) {
+      // csv writer will close under file writer
+      fileStream.close()
+      fileStream = null
+    }
+  }
+
+  def escape(str: String): String = {
+    if (str == null) {
+      ""
+    } else {
+      "\"" + str.replace("\n", "**LF**")
+        .replace("\r", "**CR**")
+        .replace("\"", "**QT**")
+        .replace(",", "**COMMA**")+ "\""
+    }
+  }
+
+  def writeRow(isLegacy: Boolean, row: Array[String]): Unit = {
+    val fullSeq = Seq(isLegacy.toString) ++ row
+    fileStream.write(fullSeq.map(escape).mkString("", ",", "\n").getBytes("UTF8"))
+  }
+}
+
+/**
+ * Used to mask customer data to avoid customer data leakage.
+ */
+object GetJsonObjectMask {
+
+  /**
+   * Used by mask data
+   */
+  private def getRetainChars: Set[Char] = {
+    val s = mutable.Set[Char]()
+    for (i <- 0 to 32) {
+      s += i.toChar
+    }
+    val others = Array[Char](
+      '{', '}', '[', ']', ',', ':', '"', '\'',
+      '\\', '/', 'b', 'f', 'n', 'r', 't', 'u',
+      '-', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'e', 'E',
+      'u', 'A', 'a', 'B', 'b', 'C', 'c', 'D', 'd', 'E', 'e', 'F', 'f',
+      't', 'r', 'u', 'e',
+      'f', 'a', 'l', 's', 'e',
+      'n', 'u', 'l', 'l',
```

Comment:
Since 'true', 'false', and 'null' are retained char-by-char rather than as a context string, there's quite a bit more coverage of the lowercase alphabet than the docs implied.  For example, "blast", "turf", "club", "nurse", and many other words are going to leak through the masking.  The docs above imply that letters l, n, r, u, t will not be retained unless they are part of the words true, false, or null.  Not must-fix, but it may be surprising to customers what may survive the masking based on the above documentation.